src/lib/structures/base/Piece.ts

```diff
@@ -0,0 +1,171 @@
+import { join } from 'path';
+import type { Client } from '../../../client/Client';
+import type { Store } from './Store';
+
+/**
+ * The common class for all pieces.
+ */
+export class Piece {
+
+	/**
+	 * The client this Piece was created with.
+	 * @since 0.0.1
+	 */
+	public readonly client: Client;
+
+	/**
+	 * The store this Piece is from.
+	 * @since 0.0.1
+	 */
+	public readonly store: Store<this>;
+
+	/**
+	 * The file location where this Piece is stored.
+	 * @since 0.0.1
+	 */
+	public readonly file: readonly string[];
+
+	/**
+	 * The base directory this Piece is stored in.
+	 * @since 0.0.1
+	 */
+	public readonly directory: string;
+
+	/**
+	 * The name of the Piece.
+	 * @since 0.0.1
+	 */
+	public readonly name: string;
+
+	/**
+	 * Whether or not the Piece is enabled.
+	 * @since 0.0.1
+	 */
+	public enabled: boolean;
+
+	/**
+	 * @since 0.0.1
+	 * @param store The store this piece is for
+	 * @param file The path from the pieces folder to the piece file
+	 * @param directory The base directory to the pieces folder
+	 * @param options The options for this piece
+	 */
+	public constructor(store: Store<Piece>, file: readonly string[], directory: string, options: PieceOptions = {}) {
```

Comment:
file should come after directory logically

Reply:
Can't `store: Store<this>` be used?

Reply:
> file should come after directory logically

https://github.com/dirigeants/klasa/blob/v1.0.0-alpha/src/lib/structures/base/Piece.js#L27-L35

> Can't `store: Store<this>` be used?

If I could, I'd have used that. However there's no `this` since it's a static method.

Reply:
> > file should come after directory logically
> 
> [dirigeants/klasa:src/lib/structures/base/Piece.js@`v1.0.0-alpha`#L27-L35](https://github.com/dirigeants/klasa/blob/v1.0.0-alpha/src/lib/structures/base/Piece.js#L27-L35) [](/dirigeants/klasa/blob/HEAD@{2020-05-09T17:41:42Z}/src/lib/structures/base/Piece.js#L27-L35)

Then rename the directory to `baseDirectory` and move it before file.. since file is the path from the baseDirectory, no?

Reply:
Fixed in f594f1a.